spec/datadog/core/crashtracking/component_spec.rb

```diff
@@ -0,0 +1,351 @@
+
+          described_class.build(settings, agent_settings, logger: logger)
+        end
+      end
+
+      context 'when missing `agent_base_url`' do
+        let(:agent_base_url) { nil }
+
+        it 'returns nil' do
+          expect(Datadog::Core::Crashtracking::TagBuilder).to receive(:call).with(settings)
+            .and_return(tags)
+          expect(Datadog::Core::Crashtracking::AgentBaseUrl).to receive(:resolve).with(agent_settings)
+            .and_return(agent_base_url)
+          expect(::Libdatadog).to receive(:ld_library_path)
+            .and_return(ld_library_path)
+          expect(::Libdatadog).to receive(:path_to_crashtracking_receiver_binary)
+            .and_return(path_to_crashtracking_receiver_binary)
+
+          expect(described_class.build(settings, agent_settings, logger: logger)).to be_nil
+        end
+      end
+
+      context 'when missing `ld_library_path`' do
+        let(:ld_library_path) { nil }
+
+        it 'returns nil' do
+          expect(Datadog::Core::Crashtracking::TagBuilder).to receive(:call).with(settings)
+            .and_return(tags)
+          expect(Datadog::Core::Crashtracking::AgentBaseUrl).to receive(:resolve).with(agent_settings)
+            .and_return(agent_base_url)
+          expect(::Libdatadog).to receive(:ld_library_path)
+            .and_return(ld_library_path)
+          expect(::Libdatadog).to receive(:path_to_crashtracking_receiver_binary)
+            .and_return(path_to_crashtracking_receiver_binary)
+
+          expect(described_class.build(settings, agent_settings, logger: logger)).to be_nil
+        end
+      end
+
+      context 'when missing `path_to_crashtracking_receiver_binary`' do
+        let(:path_to_crashtracking_receiver_binary) { nil }
+
+        it 'returns nil' do
+          expect(Datadog::Core::Crashtracking::TagBuilder).to receive(:call).with(settings)
+            .and_return(tags)
+          expect(Datadog::Core::Crashtracking::AgentBaseUrl).to receive(:resolve).with(agent_settings)
+            .and_return(agent_base_url)
+          expect(::Libdatadog).to receive(:ld_library_path)
+            .and_return(ld_library_path)
+          expect(::Libdatadog).to receive(:path_to_crashtracking_receiver_binary)
+            .and_return(path_to_crashtracking_receiver_binary)
+
+          expect(described_class.build(settings, agent_settings, logger: logger)).to be_nil
+        end
+      end
+    end
+
+    context 'instance methods' do
+      # No crash tracker process should still be running at the start of each testcase
+      around do |example|
+        wait_for { `pgrep -f libdatadog-crashtracking-receiver` }.to be_empty
+        example.run
+        wait_for { `pgrep -f libdatadog-crashtracking-receiver` }.to be_empty
+      end
+
+      describe '#start' do
+        context 'when _native_start_or_update_on_fork raises an exception' do
+          it 'logs the exception' do
+            logger = Logger.new($stdout)
```

Comment:
Consider extracting this into a `let` (it can even be shared with the `let` from `.build`, above)

Reply:
https://github.com/DataDog/dd-trace-rb/pull/3826#discussion_r1717158258